pkg/controllers/deprovisioning/helpers.go

```diff
@@ -37,6 +40,42 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func getProhibitedMachineNames(ctx context.Context, kubeClient client.Client, gates []*v1alpha5.MachineDisruptionGate) (sets.String, error) {
```

Comment:
Aren't these two functions basically always going to happen together? It seems simpler to me to reason about a function like `GetGatedMachines()` which returns a `map[v1alpha5.DisruptionMethod][]string` by looking at all the disruption gates, determining the matching machines and returning back the blocked machines for each method. WDYT?